src/app.ts

```diff
@@ -66,6 +76,20 @@ app.get("/price", cache("10 minutes"), async (req, res) => {
   res.send(price);
 });
 
+app.get("/circulating-supply", async (req, res) => {
+  if (fs.existsSync(CIRCULATING_SUPPLY_DATA_PATH)) {
+    const circulatingSupplyData = fs.readFileSync(CIRCULATING_SUPPLY_DATA_PATH);
+    res.setHeader("Content-Type", "application/json");
+    res.send(JSON.parse(circulatingSupplyData.toString()));
+
+    return;
+  }
+
+  let circulatingSupply = await getCirculatingSupply();
+  res.setHeader("Content-Type", "application/json");
+  res.send(circulatingSupply);
+});
+
 scheduleCronJob();
```

Comment:
schedule job and fetch data then start the app:
```ts
scheduleCronJob().then(() =>{
  app.listen(port, () => {
    log(`server started at http://localhost:${port}`);
  });
})
```

We could have simply used `await scheduleCronJob();` but the project needs to be configured as an es module for it to be valid to use `await` at the top level.